assets/src/components/presets.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useContext } from "react"
+import { createPreset, instantiatePreset } from "../state"
+import { StateDispatchContext } from "../contexts/stateDispatchContext"
+import { currentRouteTab, isPreset } from "../models/routeTab"
+
+const Presets = () => {
+  const [{ routeTabs }, dispatch] = useContext(StateDispatchContext)
+  const presets = routeTabs.filter(isPreset)
+
+  return (
+    <div className="m-presets-panel">
+      <button
+        onClick={() =>
+          dispatch(
+            createPreset(
+              currentRouteTab(routeTabs).uuid,
+              `Preset ${Math.floor(Math.random() * 10000)}`
```

Comment:
**suggestion (non-blocking):** I could see the state reducer (or maybe better yet a function on the `RouteTab` model…) generating the default name if one isn't provided. (But I can also imagine that this might be changing drastically soon, so I'm not really worried about it.)

Reply:
Yeah, that would arguably be the more correct way of doing it. This is very much temporary code, though, because there's going to be a modal that prompts the user for a name, which will be a required part of the preset creation process - I'll probably end up implementing it with some sort of callback that gets sent to the modal component to invoke with the specified name.